### PR TITLE
feat: allow build of different file types

### PR DESCRIPTION
**Pull request by SARCASMOOO**
## Description

Allow JavaScript files to be built for different formats.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -5,25 +5,38 @@
...
     }
   },
   "scripts": {
     "prepare": "husky",
     "commit": "git cz",
     "eslint": "eslint .",
-    "prettier": "prettier '**/*.{js,mjs,cjs,scss,sass}'",
-    "stylelint": "stylelint '**/*.scss' --allow-empty-input",
+    "prettier": "prettier 'src/**/*.{js,mjs,cjs,scss,sass}'",
+    "stylelint": "stylelint 'src/**/*.scss' --allow-empty-input",
```
Comment (NickDJM) [inline package.json]: Everything needs to be linted other than the dist directory. Do not change this.

**Review by NickDJM — CHANGES_REQUESTED** [review]
@SARCASMOOO 1 change required.

File: package.json (hunk 1 above)
Comment (SARCASMOOO) [reply package.json:32]: Okay I changed this back. I am not sure wht to do about this stylelint message though.

```
src/scss/utilities/spacing/_variables.scss
  1:1  ✖  Unknown rule @stylistic/string-quotes                      @stylistic/string-quotes
```

There are 596 of them and they are all unkowns rule messages from the sass.

Comment (SARCASMOOO) [reply package.json]: Okay I made this change.

File: vite.config.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    hmr: {
+      host: "localhost",
+    },
+  },
+  test: {
+    environment: "jsdom",
+    environmentOptions: {
+      pretendToBeVisual: true,
+    },
+  },
```
Comment (NickDJM) [inline vite.config.js:56]: Lets remove this, there's no testing suite implemented yet.

**Pull request closed without merging**